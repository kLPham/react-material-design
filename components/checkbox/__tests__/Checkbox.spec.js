import React from 'react';
import { shallow } from 'enzyme';
import Checkbox from '../index';

jest.mock('uuid', () => ({
    v4: jest.fn(() => '2e732f05-f466-4fba-acb5-2f3be58daf75'),
}));
describe('<Checkbox />', () => {
    it('should be able to render', () => {
        const label = 'My Checkbox Label';
        const checkbox = shallow(
            <Checkbox label={label} />,
        );
        expect(checkbox).toMatchSnapshot();
        expect(checkbox.find('input')).toHaveClassName('mdc-checkbox__native-control');
        expect(checkbox.find('label')).toHaveClassName('mdc-checkbox-label');
        expect(checkbox.find('label')).toHaveText(label);
    });
    it('should render as checked', () => {
        const label = 'My Checkbox Label';
        const checked = true;
        const checkbox = shallow(
            <Checkbox checked={checked} label={label} />,
        );
        expect(checkbox).toMatchSnapshot();
        expect(checkbox.find('input')).toBeChecked();
        expect(checkbox.find('input')).toHaveClassName('mdc-checkbox__native-control');
        expect(checkbox.find('label')).toHaveClassName('mdc-checkbox-label');
        expect(checkbox.find('label')).toHaveText(label);
    });
    it('should render as disabled', () => {
        const label = 'My Checkbox Label';
        const disabled = true;
        const checkbox = shallow(
            <Checkbox disabled={disabled} label={label} />,
        );
        expect(checkbox).toMatchSnapshot();
        expect(checkbox.find('input')).toBeDisabled();
        expect(checkbox.find('input')).toHaveClassName('mdc-checkbox__native-control');
        expect(checkbox.find('label')).toHaveClassName('mdc-checkbox-label');
        expect(checkbox.find('label')).toHaveText(label);
    });
});
