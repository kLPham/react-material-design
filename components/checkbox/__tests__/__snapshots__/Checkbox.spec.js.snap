// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> should be able to render 1`] = `
<FormField
  alignEnd={false}
>
  <div
    className="mdc-checkbox"
  >
    <input
      checked={false}
      className="mdc-checkbox__native-control"
      disabled={false}
      id="mdc-checkbox--2e732f05-f466-4fba-acb5-2f3be58daf75"
      onChange={[Function]}
      type="checkbox"
    />
    <div
      className="mdc-checkbox__background"
    >
      <svg
        className="mdc-checkbox__checkmark"
        version="1.1"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          className="mdc-checkbox__checkmark__path"
          d="M1.73,12.91 8.1,19.28 22.79,4.59"
          fill="none"
          stroke="white"
        />
      </svg>
      <div
        className="mdc-checkbox__mixedmark"
      />
    </div>
  </div>
  <label
    className="mdc-checkbox-label"
    htmlFor="mdc-checkbox--2e732f05-f466-4fba-acb5-2f3be58daf75"
  />
</FormField>
`;

exports[`<Checkbox /> should be able to render with a label 1`] = `
<FormField
  alignEnd={false}
>
  <div
    className="mdc-checkbox"
  >
    <input
      checked={false}
      className="mdc-checkbox__native-control"
      disabled={false}
      id="mdc-checkbox--2e732f05-f466-4fba-acb5-2f3be58daf75"
      onChange={[Function]}
      type="checkbox"
    />
    <div
      className="mdc-checkbox__background"
    >
      <svg
        className="mdc-checkbox__checkmark"
        version="1.1"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          className="mdc-checkbox__checkmark__path"
          d="M1.73,12.91 8.1,19.28 22.79,4.59"
          fill="none"
          stroke="white"
        />
      </svg>
      <div
        className="mdc-checkbox__mixedmark"
      />
    </div>
  </div>
  <label
    className="mdc-checkbox-label"
    htmlFor="mdc-checkbox--2e732f05-f466-4fba-acb5-2f3be58daf75"
  >
    My Checkbox Label
  </label>
</FormField>
`;

exports[`<Checkbox /> should render as checked 1`] = `
<FormField
  alignEnd={false}
>
  <div
    className="mdc-checkbox"
  >
    <input
      checked={true}
      className="mdc-checkbox__native-control"
      disabled={false}
      id="mdc-checkbox--2e732f05-f466-4fba-acb5-2f3be58daf75"
      onChange={[Function]}
      type="checkbox"
    />
    <div
      className="mdc-checkbox__background"
    >
      <svg
        className="mdc-checkbox__checkmark"
        version="1.1"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          className="mdc-checkbox__checkmark__path"
          d="M1.73,12.91 8.1,19.28 22.79,4.59"
          fill="none"
          stroke="white"
        />
      </svg>
      <div
        className="mdc-checkbox__mixedmark"
      />
    </div>
  </div>
  <label
    className="mdc-checkbox-label"
    htmlFor="mdc-checkbox--2e732f05-f466-4fba-acb5-2f3be58daf75"
  >
    My Checkbox Label
  </label>
</FormField>
`;

exports[`<Checkbox /> should render as disabled 1`] = `
<FormField
  alignEnd={false}
>
  <div
    className="mdc-checkbox"
  >
    <input
      checked={false}
      className="mdc-checkbox__native-control"
      disabled={true}
      id="mdc-checkbox--2e732f05-f466-4fba-acb5-2f3be58daf75"
      onChange={[Function]}
      type="checkbox"
    />
    <div
      className="mdc-checkbox__background"
    >
      <svg
        className="mdc-checkbox__checkmark"
        version="1.1"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          className="mdc-checkbox__checkmark__path"
          d="M1.73,12.91 8.1,19.28 22.79,4.59"
          fill="none"
          stroke="white"
        />
      </svg>
      <div
        className="mdc-checkbox__mixedmark"
      />
    </div>
  </div>
  <label
    className="mdc-checkbox-label"
    htmlFor="mdc-checkbox--2e732f05-f466-4fba-acb5-2f3be58daf75"
  >
    My Checkbox Label
  </label>
</FormField>
`;
